grommet/grommet
# Add ability to type form component

Signed-off-by: Alexander Mattoni <5110855+mattoni@users.noreply.github.com>



#### What does this PR do?

This PR adds the ability to type Form components with the values expected.

```
// event.values is typed to InputTypes now.
<Form<InputTypes> onSubmit={event => console.log(event.values)} />
```

#### Where should the reviewer start?

`src/js/components/Form/index.d.ts`

#### What testing has been done on this PR?

Confirmed the prop types work on <Form> components, and that it works as it used to if not typed.

#### How should this be manually tested?

Create a <Form> component and type it with an interface of expected input values. Then, ensure onSubmit event.values is typed properly.

#### Any background context you want to provide?

#### What are the relevant issues?

#4840 

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?

#### Should this PR be mentioned in the release notes?

Probably

#### Is this change backwards compatible or is it a breaking change?
Backward compatible
